gen/AArch64Compile_gen.ml

```diff
@@ -1729,15 +1729,18 @@ module Make(Cfg:Config) : XXXCompile_gen.S =
     | None,_ -> Warn.fatal "AArchCompile.emit_access"
     | Some d,Code lab ->
         begin match d,e.C.atom with
-        | R,None ->
+        | R,Some (Instr, None) ->
             let r,init,cs,st = LDR.emit_fetch st p init lab in
             Some r,init,cs,st
-        | W,(None | Some (Instr, None)) ->
-            let init,cs,st = STR.emit_store_nop st p init lab in
-            None,init,cs,st
-        | R, Some (Instr, None) ->
+        (* Plain read from an instruction label is currently not supported, 
+           but will be implemented in a future patch
+        | R, None ->
             let r,init,cs,st = LDR.emit_load st p init lab in
             Some r,init,cs,st
+        | W, Some(Instr, None)  *)
+        | W, None ->
+            let init,cs,st = STR.emit_store_nop st p init lab in
+            None,init,cs,st
```

Comment:
It should be relatively easy to implement additional  `emit_store_nop` function, performing store release, and maybe  atomic stores. Maybe for a later PR?

Reply:
yes, that makes a lot of sense! I will definitely have a look at that after this PR. Thank you for the suggestion!